ddl/multi_schema_change.go

```diff
@@ -93,19 +93,33 @@ func onMultiSchemaChange(w *worker, d *ddlCtx, t *meta.Meta, job *model.Job) (ve
 			proxyJob := cloneFromSubJob(job, sub)
 			ver, err = w.runDDLJob(d, t, proxyJob)
 			mergeBackToSubJob(proxyJob, sub)
-			handleRevertibleException(job, sub, i, proxyJob.Error)
+			handleRevertibleException(job, sub, proxyJob.Error)
 			return ver, err
 		}
 		// All the sub-jobs are non-revertible.
 		job.MultiSchemaInfo.Revertible = false
 		// Step the sub-jobs to the non-revertible states all at once.
-		for _, sub := range job.MultiSchemaInfo.SubJobs {
+		for i, sub := range job.MultiSchemaInfo.SubJobs {
 			if isFinished(sub) {
 				continue
 			}
 			proxyJob := cloneFromSubJob(job, sub)
 			ver, err = w.runDDLJob(d, t, proxyJob)
 			mergeBackToSubJob(proxyJob, sub)
+			if err != nil {
+				job.MultiSchemaInfo.Revertible = true
```

Comment:
We can set this to `false` after all the sub-job run successfully.

Reply:
IMO, we can't do this. Because we use `if job.MultiSchemaInfo.Revertible` in many sub-jobs to verify we should return or not.

Reply:
The meaning of `MultiSchemaInfo.Revertible` is different between the parent job and the sub-job.

- For parent job: all the sub-jobs are non-revertible
- For sub-job: the current job is about to be non-revertible

Here we refer to the parent job.